src/features/click-to-load/components/ctl-login-button.js

```diff
@@ -0,0 +1,169 @@
+import cssVars from '../assets/shared.css'
+import css from '../assets/ctl-login-button.css'
+import { logoImg } from '../ctl-assets'
+
+/**
+ * @typedef LearnMoreParams - "Learn More" link params
+ * @property {string} readAbout - "Learn More" aria-label text
+ * @property {string} learnMore - "Learn More" link text
+ */
+
+export class DDGCtlLoginButton extends HTMLElement {
+    static CUSTOM_TAG_NAME = 'ddg-ctl-login-btn'
+
+    /**
+     * @param {object} params - Params for building a custom element with
+     *                          a placeholder for a blocked login button
+     * @param {boolean} params.devMode - Used to create the Shadow DOM on 'open'(true) or 'closed'(false) mode
+     * @param {string} params.label - Button text
+     * @param {string} params.logoIcon - Logo image to be displayed in the Login Button to the left of the label text
+     * @param {string} params.hoverText - Text for popover on button hover
+     * @param {boolean=} params.useSlimCard - Flag for using less padding on card (ie YT CTL on mobile)
+     * @param {HTMLElement} params.originalElement - The original element this placeholder is replacing.
+     * @param {LearnMoreParams} params.learnMore - Localized strings for "Learn More" link.
+     * @param {(originalElement: HTMLIFrameElement | HTMLElement, replacementElement: HTMLElement) => (e: any) => void} params.onClick
+     */
+    constructor (params) {
+        super()
+        this.params = params
+        /**
+         * Create the shadow root, closed to prevent any outside observers
+         * @type {ShadowRoot}
+         */
+        const shadow = this.attachShadow({
+            mode: this.params.devMode ? 'open' : 'closed'
+        })
+
+        /**
+         * Add our styles
+         * @type {HTMLStyleElement}
+         */
+        const style = document.createElement('style')
+        style.innerText = cssVars + css
+
+        /**
+         * Create the Facebook login button
+         * @type {HTMLDivElement}
+         */
+        const loginButton = this.createLoginButton()
+
+        /**
+         * Setup the click handlers
+         */
+        this.setupEventListeners(loginButton)
+
+        /**
+         * Append both to the shadow root
+         */
+        shadow.appendChild(loginButton)
+        shadow.appendChild(style)
+    }
+
+    /**
+     * Creates a placeholder Facebook login button. When clicked, a warning dialog
+     * is displayed to the user. The login flow only continues if the user clicks to
+     * proceed.
+     * @returns {HTMLDivElement}
+     */
+    createLoginButton = () => {
+        const { label, hoverText, logoIcon, learnMore } = this.params
+
+        const { popoverStyle, arrowStyle } = this.calculatePopoverPosition()
+
+        const container = document.createElement('div')
+        // Add our own styles and inherit any local class styles on the button
+        container.classList.add('ddg-fb-login-container')
+
+        container.innerHTML = html`
```

Comment:
Assigning to innerHTML like this makes me a little nervous, especially since we're putting translated strings in there. Has potential to be a security vulnerability if a translation sneaks in with some malicious HTML for example. This might be fine and there might not be a better way, just thought I should raise it.

Reply:
Actually the `html` tagged template function used here will handle the escaping of any content injected via `${...}`

Reply:
Cool OK, so for this one we don't need to use any localised strings that are supposed to contain HTML?

Reply:
That's a good point. If the localization has HTML, I guess it wouldn't work. So far, the translations for the elements in these web components didn't have an HTML embedded in them. I'm aware that the YouTube Preview has one, although the mobile layouts and copy planned for it I believe don't have it.

Should we make an effort to avoid HTML in the translations? Perhaps either break down the string when possible or give preference to a placeholder to extract what needs to be wrapped in a HTML tag.

Reply:
@franfaccin sounds likes a good approach for now. If it becomes unavoidable then we can make the `html` tag work - but no need right now if we can avoid :)

Reply:
> Should we make an effort to avoid HTML in the translations?

Well, in the past I've seen translation systems that allow translated strings to contain a subset of HTML tags (e.g. <b>, <i>, <em>, <a>). All attributes were stripped from those tags in the translation and the link targets etc were populated by the system at runtime. I think sometimes you do need HTML in translations, to handle the "Learn more" case properly.

I guess this is off-topic for this review though, so feel free to resolve. I guess we can just punt this, but it would be good if our translation / HTML templating systems could handle it in the future.

Reply:
I like the idea of a subset of tags without any properties. One of the issues of having HTML in the translated copies, besides possible injections, is maintaining the HTML. We might need to change the CSS, add/remove attributes, change href, etc, and that becomes hard to do when they're all embedded in the translated string. So having a placeholder in the string to extract what needs to wrapped in HTML or some subset of simple tags that can be repopulated with the necessary attributes sound like good ideas to consider.

But like you said, this is not our case in the project yet 😅 But I agree we will need to figure out how to make it work best once that starts happening.